osbuild-composer.spec

```diff
@@ -294,7 +293,7 @@ cd $PWD/_build/src/%{goipath}
 
 %package core
 Summary:    The core osbuild-composer binary
-Requires:   %{name}-dnf-json = %{version}-%{release}
+Requires:   osbuild-depsolve-dnf >= 98
```

Comment:
Please see https://github.com/osbuild/osbuild/pull/1524#pullrequestreview-1812744808

You should add the following lines to ensure that the `osbuild-composer-dnf-json` will get removed on upgrade:
```spec
Provides: %{name}-dnf-json = %{version}-%{release}
Obsoletes: %{name}-dnf-json < %{version}-%{release}
```

However, the tricky part is that also the `worker` subpackage had the requires on the old `dnf-json` sub-package and the `-core` sub-package may not be installed on workers. I'm not sure what will happen if these lines are added to both sub-packages. You'll have to test this to ensure that it works in all scenarios:

1. When `osbuild-composer` package is installed, which would imply `osbuild-composer-core` to be installed as well. This is the on-prem scenario.
2. When only the `osbuild-composer-worker` package is installed. This is the "remote worker" scenario that we use e.g. in the service.

Reply:
Thank you, I tested both cases and `osbuild-composer-dnf-json` was successfully replaced in both cases:
```
 osbuild-composer-core                                                  x86_64                             100-1.20240202git8a775a3.fc39                                    @commandline                              10 M
     replacing  osbuild-composer-dnf-json.x86_64 99-1.fc39
 osbuild-composer-worker                                                x86_64                             100-1.20240202git8a775a3.fc39                                    @commandline                              14 M
     replacing  osbuild-composer-dnf-json.x86_64 99-1.fc39
```
and
```
 osbuild-composer-worker                                   x86_64                                   100-1.20240202git8a775a3.fc39                                     @commandline                                    14 M
     replacing  osbuild-composer-dnf-json.x86_64 99-1.fc39
Installing dependencies:
 osbuild-depsolve-dnf                                      noarch                                   104-1.fc39                                                        updates                                         49 k

```